src/mono/wasm/runtime/library_mono.js

```diff
@@ -2252,6 +2359,36 @@ var MonoSupportLib = {
 		};
 		debugger;
 	},
+
+	mono_wasm_asm_loaded: function (assembly_name, assembly_ptr, assembly_len, pdb_ptr, pdb_len) {
+		// Only trigger this codepath for assemblies loaded after app is ready
+		if (MONO.mono_wasm_runtime_is_ready !== true)
+			return;
+
+		if (!this.mono_wasm_assembly_already_added)
+			this.mono_wasm_assembly_already_added = Module.cwrap ("mono_wasm_assembly_already_added", 'number', ['string']);
+
+		// And for assemblies that have not already been loaded
+		const assembly_name_str = assembly_name !== 0 ? Module.UTF8ToString(assembly_name).concat('.dll') : '';
+		if (this.mono_wasm_assembly_already_added(assembly_name_str))
```

Comment:
What about: `foo.dll, assembly name "Bar"`? We check for `Bar.dll`. IIUC, this doesn't break anything though, since we'll just end up sending this assembly anyway, and the proxy can dedupe it based on the assembly_name.

I think @lewing was suggesting that may be we can compare the image/data pointer? Would that make sense? Is that accessible in driver.c?

Reply:
> I think @lewing was suggesting that may be we can compare the image/data pointer? Would that make sense? Is that accessible in driver.c?

Yeah, there is an `entry->assembly.data` property but the addresses don't seem to be the same there so the comparison always fails.